source/Calamari/Kubernetes/Integration/HelmCli.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+
+namespace Calamari.Kubernetes.Integration
+{
+    public class HelmCli : CommandLineTool
+    {
+        public HelmCli(ILog log, ICommandLineRunner commandLineRunner, string workingDirectory, Dictionary<string, string> environmentVars)
+            : base(log, commandLineRunner, workingDirectory, environmentVars)
+        {
+            ExecutableLocation = "helm";
+        }
+
+        readonly List<string> builtInArguments = new List<string>();
+
+        public HelmCli WithExecutable(string customExecutable)
+        {
+            ExecutableLocation = customExecutable;
+            return this;
+        }
+
+        public HelmCli WithExecutable(IVariables variables)
+        {
+            var helmExecutable = variables.Get(SpecialVariables.Helm.CustomHelmExecutable);
+            if (string.IsNullOrWhiteSpace(helmExecutable))
+            {
+                return this;
+            }
+
+            if (variables.GetIndexes(PackageVariables.PackageCollection)
```

Comment:
Any other variables we need to handle? Or is this all?

Reply:
The variables being passed here are all the variables that were passed down in the deployment, which I believe should be the lot of them.

This is eq to what we are currently doing for the helm upgrade step